src/shared/loop-util.c

```diff
@@ -167,6 +188,17 @@ static int loop_configure(
         }
 
         if (*try_loop_configure) {
+                /* Acquire uevent seqnum immediately before attaching the loopback device. This allows
+                 * callers to ignore all uevents with a seqnum before this one, if they need to associate
+                 * uevent with this attachment. Doing so isn't race-free though, as uevents that happen in
+                 * the window between this reading of the seqnum, and the LOOP_CONFIGURE call might still be
+                 * mistaken as originating from our attachment, even though might be caused by an earlier
+                 * use. But doing this at least shortens the race window a bit. */
+                r = get_current_uevent_seqnum(&seqnum);
+                if (r < 0)
+                        return r;
+                timestamp = now(CLOCK_MONOTONIC);
```

Comment:
Why these are called here, instead of just before `return`?
```
if (ret_seqnum_not_before) {
        r = get_current_uevent_seqnum(ret_seqnum_not_before)
        if (r < 0)
                return r;
}
if (ret_timestamp_not_before)
        *ret_timestamp_not_before = now(CLOCK_MONOTONIC);
return 0;
```

Reply:
see the comments. we want these timestamps/seqnums that were current immediately *before* we attach the file to the loopback file. If we'd take them afterwards we might take them at a point where the kernel already fired off some uevents, and then they wouldn't work as boundary to filter db entries/uevents with anymore.

Reply:
Ah, I see. Now I understand the comment. Thank you.